### PR TITLE
Add smart contract evaluation

### PR DESCRIPTION
**Pull request by jkrvivian**

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -408,10 +410,43 @@ \subsection{Launch Data Products}
     \label{fig:launching_product}
 \end{figure}
 
+\begin{lstlisting}[caption={Functions of hased key update and certified key update}, label={lst:key_certification}, frame=single]
+	function addHashedKey(
```
Comment (jserv) [inline paper.tex]: Use 4-spaces for indention.

File: paper.tex [hunk 2]
```diff
@@ -399,7 +401,7 @@ \subsection{Prerequisite}
...
+The details of data product, such as data price, MAM channel/endpoint ID and the length of data stream are listed on the Product Contract. As regard to the encryption key of data product, data providers send the hash digest of encryption key to a broker, who will then sign it and written on the Product Contract. This provides an approach for data consumers to verify the encryption key that avoid data providers dishonest toward data consumers with a wrong one. The workflow is shown in Fig.~\ref{fig:launching_product} and the certification codes in Product Contract is in Listing.~\ref{lst:key_certification}. Finally, the data product is launched and data providers can start uploading data.
```
Comment (jserv) [inline paper.tex]: Check the grammar: written on the Product Contract?

File: paper.tex [hunk 3]
```diff
@@ -553,6 +614,68 @@ \subsubsection{Messages Publishment}
...
 
+\subsection{Smart Contract}
+In this experiment, we investigate the cost of executing Product Contract with Ethereum web browser based IDE Remix that we can define the minimum cost, which data providers should pay in order to launch a product and processing a trade.  On top of this evaluation, data providers can consider how to set the price to eventually generate profits. The estimated cost is in Ether, which is 238 USD at the time of writing, yet the price is dynamic from time to time. To conquer this condition, the authors in \cite{MindMyValue} suggested to analyze with a fixed range of Gas price,  yet the main drawback of low Gas price is the increase of time required before a transaction is validated. In the report of Ethereum network\cite{ethereumChart} in 2020 (January to June), we set this range from 7.3 Gwei to 26.5 Gwei, which is $7.3 \times 10^{-9}$ and $2.65 \times 10^{-8}$ Ether respectively, the minimum average and average gas price. The Gas usage estimation is shown in Table.~\ref{tab:gas}.
```
Comment (jserv) [inline paper.tex]: gas 價格以何時為主？

Comment (jkrvivian) [reply paper.tex]: 2020 (January to June)

Comment (jkrvivian) [reply paper.tex]: 我修正敘述

**Pull request merged**